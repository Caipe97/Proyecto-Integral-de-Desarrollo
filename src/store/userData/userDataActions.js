import {
  LOGOUT,
  REFRESH_PAGE,

  LOGIN_OR_REGISTER_PENDING,
  LOGIN_OR_REGISTER_SUCCESS,
  LOGIN_OR_REGISTER_FAILED,

  RESET_PASSWORD_PENDING,
  RESET_PASSWORD_SUCCESS,
  RESET_PASSWORD_FAILED
 } from './userDataConstants'

export const logout = () => ({ 
  type: LOGOUT
})

export const refreshPage = () => ({ 
  type: REFRESH_PAGE
})

export const login = (email, password) => (dispatch) => {
  dispatch({ type: LOGIN_OR_REGISTER_PENDING })
  return(
    fetch(`https://jma-test-app.herokuapp.com/api/users/login`, {
      method: 'POST',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify({
        email: email,
        password: password
      })
    })
  )
  .then(response => response.json())
  .then(userData => dispatch({ type: LOGIN_OR_REGISTER_SUCCESS, payload: userData }))
  .catch(error => dispatch({ type: LOGIN_OR_REGISTER_FAILED, payload: error }))
}

export const register = (name, surname, email, password, gender, birthday, weight, height) => (dispatch) => {
  dispatch({ type: LOGIN_OR_REGISTER_PENDING });
  return(
    fetch(`https://jma-test-app.herokuapp.com/api/users/register`, {
      method: 'POST',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify({
        name: name,
        surname: surname,
        email: email,
        password: password,
        gender: gender,
        birthday: birthday,
        weight: weight,
        height: height
      })
    })
  )
  .then(response => response.json())
  .then(userData => dispatch({ type: LOGIN_OR_REGISTER_SUCCESS, payload: userData }))
  .catch(error => dispatch({ type: LOGIN_OR_REGISTER_FAILED, payload: error }))
}

export const resetPassword = (userId, password) => (dispatch) => {
  dispatch({ type: RESET_PASSWORD_PENDING });
  return(
    fetch(`https://jma-test-app.herokuapp.com/api/users?userId=${userId}`, {
      method: 'PUT',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify({
        password: password
      })
    })
  )
  .then(response => response.json())
  .then(userData => dispatch({ type: RESET_PASSWORD_SUCCESS, payload: userData }))
  .catch(error => dispatch({ type: RESET_PASSWORD_FAILED, payload: error }))
}