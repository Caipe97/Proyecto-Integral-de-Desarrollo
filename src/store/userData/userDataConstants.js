export const LOGOUT = 'LOGOUT';

export const LOGIN_OR_REGISTER_PENDING = 'LOGIN_OR_REGISTER_PENDING';
export const LOGIN_OR_REGISTER_SUCCESS = 'LOGIN_OR_REGISTER_SUCCESS';
export const LOGIN_OR_REGISTER_FAILED = 'LOGIN_OR_REGISTER_FAILED';

export const RESET_PASSWORD_PENDING = 'RESET_PASSWORD_PENDING';
export const RESET_PASSWORD_SUCCESS = 'RESET_PASSWORD_SUCCESS';
export const RESET_PASSWORD_FAILED = 'RESET_PASSWORD_FAILED';