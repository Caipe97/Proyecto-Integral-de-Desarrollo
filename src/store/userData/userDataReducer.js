import {
  LOGOUT,

  LOGIN_OR_REGISTER_PENDING,
  LOGIN_OR_REGISTER_SUCCESS,
  LOGIN_OR_REGISTER_FAILED,

  RESET_PASSWORD_PENDING,
  RESET_PASSWORD_SUCCESS,
  RESET_PASSWORD_FAILED
 } from './userDataConstants';

const initialStateUserData = {
  id: 0,
  name: '',
  surname: '',
  email: '',
  password: '',
  gender: '',
  birthday: '',
  weight: '',
  height: '',
  isPending: false
}

export const userDataReducer = (state=initialStateUserData, action={}) => {
  switch (action.type) {
    case LOGOUT:
      return {
        id: 0,
        name: '',
        surname: '',
        email: '',
        password: '',
        gender: '',
        birthday: '',
        weight: '',
        height: '',
        isPending: false
      }
    case LOGIN_OR_REGISTER_PENDING:
      return {
        ...state,
        isPending: true
      }
    case LOGIN_OR_REGISTER_SUCCESS:
      return {
        ...state,
        id: action.payload.id,
        name: action.payload.name,
        surname: action.payload.surname,
        email: action.payload.email,
        password: action.payload.password,
        gender: action.payload.gender,
        birthday: action.payload.birthday,
        weight: action.payload.weight,
        height: action.payload.height,
        isPending: false
      }
    case LOGIN_OR_REGISTER_FAILED:
      return {
        ...state,
        error: action.payload,
        isPending: false
      }
    case RESET_PASSWORD_PENDING:
      return {
        ...state,
        isPending: true
      }
    case RESET_PASSWORD_SUCCESS:
      return {
        ...state,
        isPending: false
      }
    case RESET_PASSWORD_FAILED:
      return {
        ...state,
        isPending: false,
        error: action.payload
      }
    default:
      return state
  }
}