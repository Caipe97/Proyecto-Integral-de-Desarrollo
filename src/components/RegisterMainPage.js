import React, { Component }  from 'react';
import { Link } from "react-router-dom";
import TextField from '@material-ui/core/TextField';
import '../pages/Register/Register.css';
import logo2 from '../images/2pages.jpg';
import Header from '../components/Header';
import Footer from '../components/Footer';
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import { LOGIN_OR_REGISTER_SUCCESS } from '../store/userData/userDataConstants';

//const bcrypt = require('bcryptjs');
/* eslint-disable */
class RegisterMainPage extends Component{
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      surname: '',
      email: '',
      password: '',
      gender: '',
      birthday: '',
      weight: '',
      height: '',
      errorMessage: ''
    };
  }

  handleChange = event => {
    this.setState({
      ...this.state,
      [event.target.name]: event.target.value,
      errorMessage: ''
    })
  };

  handleChangeBirthday = date => {
    console.log(date)
    this.setState({
      ...this.state,
      birthday: date,
      errorMessage: ''
    })
  };

  handleSubmit = async event => {
    event.preventDefault();
    const birhtdayString = this.state.birthday.toString().substring(4, 24);
    //const valueCryptoPass=bcrypt.hashSync(this.state.password);
    if(this.validateAll()){
      const data = await this.props.onRegister(this.state.name, this.state.surname, this.state.email, this.state.password, this.state.gender, birhtdayString, this.state.weight, this.state.height);
      this.setState({ name: '', surname: '', email: '', password: '', gender: '', birthday: '', weight: '', height: '' });
      if(data.type === LOGIN_OR_REGISTER_SUCCESS){
        this.props.history.push("/profile");
      } else {
        this.setState({...this.state, errorMessage: 'Ya existe un usuario con ese email'});
      }
    } else {
      this.setState({...this.state, errorMessage: 'No completaste los campos correctamente'});
    }
  };

  validateAll = () => {
    if(this.validateName(this.state.name) && 
      this.validateSurname(this.state.surname) && 
      this.validateEmail(this.state.email) && 
      this.validatePassword(this.state.password) &&
      this.validateGender(this.state.gender) &&
      this.validateWeight(this.state.weight) &&
      this.validateHeight(this.state.height)){
        return true;
    }
    else{
        return false;
    }
  }

  validateName = (name) => { //cannot be empty and only letters
    const expression = /^[a-z ,.'-]+$/i 
    return expression.test(String(name).toLowerCase())
  }

  validateSurname = (surname) => { //cannot be empty and only letters
    const expression = /^[a-z ,.'-]+$/i 
    return expression.test(String(surname).toLowerCase())
  }

  validateEmail = (email) => { //cannot be empty and valid mail
    const expression = /(?!.*\.{2})^([a-z\d!#$%&'*+\-\/=?^_`{|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+(\.[a-z\d!#$%&'*+\-\/=?^_`{|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+)*|"((([\t]*\r\n)?[\t]+)?([\x01-\x08\x0b\x0c\x0e-\x1f\x7f\x21\x23-\x5b\x5d-\x7e\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|\\[\x01-\x09\x0b\x0c\x0d-\x7f\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))*(([\t]*\r\n)?[\t]+)?")@(([a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|[a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF][a-z\d\-._~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]*[a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])\.)+([a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|[a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF][a-z\d\-._~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]*[a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])\.?$/i;
    return expression.test(String(email).toLowerCase())
  }
  
  validatePassword = (password) => { //cannot be empty
    const expression = /(.|\s)*\S(.|\s)*/ 
    return expression.test(String(password).toLowerCase())
  }

  validateGender = (gender) => { //cannot be empty and only letters
    const expression = /^[a-z ,.'-]+$/i 
    return expression.test(String(gender).toLowerCase())
  }

  validateWeight = (weight) => { //cannot be empty and only numbers
    const expression = /^\d+$/
    return expression.test(String(weight).toLowerCase())
  }

  validateHeight = (height) => { //cannot be empty and only numbers
    const expression = /^\d+$/
    return expression.test(String(height).toLowerCase())
  }

  render() {
    return (
      <div className="contenedorRe">
        <Header userId={this.props.userId} onLogout={this.props.onLogout} history={this.props.history}/>
        <div className="contenidoRe">
          <div className="col1">
          <img src={logo2} alt="imagen tracking" style={{ width:240,height:240 }} />
          
          <h1 style={{ left: '30px', color:'#002626', fontFamily: 'Arial'}}>Nutriva</h1>
          </div>
          
          <div className="col2" style ={{justifyContent:'center',alignContent:'center',alignItems:'center',textAlign:'center'}}>
          <br></br>
          <br></br>
              <h1 style={{ left: '30px', color:'#002626', fontFamily: 'Arial'}}>Buscamos</h1>
              
              <br></br>
              <p>Contribuir al desarrollo integral de los clientes mediante el diseño y la aplicación de un conjunto de servicios que fortalezcan la salud física y mental.</p>
              <br></br>
              <p>Ayudar a nuestros socios a crear y mantener una excelente condición física a través de la nutrición y programas de entrenamiento.</p>
              <br></br>
              <p>Inculcar en niños, jóvenes y adultos, el valor de la salud y el ejercicio.</p>
              <br></br>
              {/* <p>Mejorar la calidad de vida de las comunidades a través de la filosofía del ejercicio, programas y productos de nutrición.</p> */}

              <p>Participar proactivamente en el bienestar de las personas en su entorno social y laboral.</p>
              <br></br>
              <p>Brindar un servicio único y facilitar las herramientas del éxito.</p>

          </div>
          
        </div>
        <div className="sidebar">
          <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', flexDirection: 'column' }}>

            <div style={{ marginTop: 45 }}>
              <h1 className='f1' style={{color:'#002626', fontFamily: 'Arial'}}>Register</h1>
            </div>
  
            <form style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}>
              <TextField label="Name" name='name' type='name' value={this.state.name} onChange={this.handleChange} required />
              <TextField label="Surname" name='surname' type='surname' value={this.state.surname} onChange={this.handleChange} required />
              <TextField label="Email" name='email' type='email' value={this.state.email} onChange={this.handleChange} required />
              <TextField label="Password" name='password' type='password' value={this.state.password} onChange={this.handleChange} required />
              <div style={{margin: 10}}>
                <input type="radio" id="masculino" name="gender" value="M" onChange={this.handleChange}/>
                <label>Masculino</label>
                <input type="radio" id="femenino" name="gender" value="F" onChange={this.handleChange}/>
                <label>Femenino</label>
              </div>
  
              <DatePicker
                showTimeSelect
                name='date'
                selected={this.state.birthday}
                onChange={(date) => this.handleChangeBirthday(date)}
                dateFormat="dd-MM-yyyy h:mm aa"
                placeholderText='birthday'
              />
              <TextField label="Weight" name='weight' type='weight' value={this.state.weight} onChange={this.handleChange} required />
              <TextField label="Height" name='height' type='height' value={this.state.height} onChange={this.handleChange} required />
              <button onClick={this.handleSubmit} className='button'>Registrarse</button>
              <p>{this.state.errorMessage}</p>
              <p style={{ marginBlock: '0em', marginTop: '10%' }}>Si ya tenes una cuenta?</p>
              <Link to="/" style={{ color: 'black' }}>Login</Link>
            </form>
          </div>
        </div>
        <Footer />
      </div>
  
    );
  }
}

export default RegisterMainPage
