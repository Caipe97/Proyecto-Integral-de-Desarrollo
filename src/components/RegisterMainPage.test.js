import React from 'react';
import { shallow } from 'enzyme';
import RegisterMainPage from './RegisterMainPage';
import { LOGIN_OR_REGISTER_SUCCESS } from '../store/userData/userDataConstants';

describe('RegisterMainPage tests', () => {
    const historyMock = { push: jest.fn() };
    const preventDefault = { preventDefault: jest.fn() }
    let wrapper;
    const onRegisterReturnedDataSuccess = { 
        type: LOGIN_OR_REGISTER_SUCCESS,
        payload: {
            id: 1,
            name: 'julian',
            surname: 'livrone',
            email: 'julianlivrone@gmail.com',
            password: 'asd',
            gender: 'masculino',
            birthday: '2010-01-21',
            weight: 90,
            height: 170
        }
    };

    beforeEach(() => {
        let mockProps = {
            userDataReducer: {
                id: 0,
                name: '',
                surname: '',
                email: '',
                password: '',
                gender: '',
                birthday: '',
                weight: '',
                height: '',
                isPending: false
            },
            onRegister: jest.fn().mockReturnValueOnce(onRegisterReturnedDataSuccess),
            onLogout: jest.fn(),
            preventDefault:  jest.fn()
          };
        wrapper = shallow(<RegisterMainPage {...mockProps} history={historyMock}/>);
        wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: 'julian' } });
        wrapper.find('[name="surname"]').at(0).simulate('change', { target: { name: 'surname', value: 'livrone' } });
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: 'julianlivrone@gmail.com' } });
        wrapper.find('[name="password"]').at(0).simulate('change', { target: { name: 'password', value: 'asd' } });
        wrapper.find('[name="gender"]').at(0).simulate('change', { target: { name: 'gender', value: 'M' } });
        wrapper.find('[name="date"]').at(0).simulate('change', 'Sat Sep 18 2021 11:00:00 GMT-0300 (hora estándar de Argentina)');
        wrapper.find('[name="weight"]').at(0).simulate('change', { target: { name: 'weight', value: '90' } });
        wrapper.find('[name="height"]').at(0).simulate('change', { target: { name: 'height', value: '170' } });
    })

    
    it('renders RegisterMainPage without crashing', () => {
        wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: '' } });
        wrapper.find('[name="surname"]').at(0).simulate('change', { target: { name: 'surname', value: '' } });
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: '' } });
        wrapper.find('[name="password"]').at(0).simulate('change', { target: { name: 'password', value: '' } });
        wrapper.find('[name="gender"]').at(0).simulate('change', { target: { name: 'gender', value: '' } });
        wrapper.find('[name="date"]').at(0).simulate('change', '');
        wrapper.find('[name="weight"]').at(0).simulate('change', { target: { name: 'weight', value: '' } });
        wrapper.find('[name="height"]').at(0).simulate('change', { target: { name: 'height', value: '' } });
        expect(wrapper).toMatchSnapshot();
    })
    
    it('handleChange should update the state correctly when completing the inputs', () => {
        expect(wrapper.instance().state).toEqual({
            name: 'julian',
            surname: 'livrone',
            email: 'julianlivrone@gmail.com',
            password: 'asd',
            gender: 'M',
            birthday: 'Sat Sep 18 2021 11:00:00 GMT-0300 (hora estándar de Argentina)',
            weight: '90',
            height: '170'
        });
    })

    it('handleSubmit should update the state to its initial state (empty)', async () => {
        await wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().state).toEqual({
            name: '',
            surname: '',
            email: '',
            password: '',
            gender: '',
            birthday: '',
            weight: '',
            height: ''
        });
    })

    it('handleSubmit should push to profile when fetching the userData when all inputs are valid', async () => {
        await wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(1);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(1);
        expect(historyMock.push.mock.calls[0]).toEqual(['/profile']);
    })

    it('handleSubmit should not push to profile because name is empty', () => {
        wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: '' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because name contains numbers', () => {
        wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: 'julian123' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because name contains special characters', () => {
        wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: 'julian@#$' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because surname contains numbers', () => {
        wrapper.find('[name="surname"]').at(0).simulate('change', { target: { name: 'surname', value: 'livrone123' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because surname contains special characters', () => {
        wrapper.find('[name="surname"]').at(0).simulate('change', { target: { name: 'surname', value: 'livrone@#$' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because email is not valid 1', () => {
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: '' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because email is not valid 2', () => {
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: 'julianlivrone' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because email is not valid 3', () => {
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: 'julianlivrone@' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because email is not valid 4', () => {
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: 'julianlivrone@gmail' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because email is not valid 5', () => {
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: 'julianlivrone@gmail.' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because email is not valid 5', () => {
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: 'julianlivrone@gmail.' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because password is empty', () => {
        wrapper.find('[name="password"]').at(0).simulate('change', { target: { name: 'password', value: '' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because gender is empty', () => {
        wrapper.find('[name="gender"]').at(0).simulate('change', { target: { name: 'gender', value: '' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because weight is empty', () => {
        wrapper.find('[name="weight"]').at(0).simulate('change', { target: { name: 'weight', value: '' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because weight contains letters', () => {
        wrapper.find('[name="weight"]').at(0).simulate('change', { target: { name: 'weight', value: '123abc' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because weight contains special characters', () => {
        wrapper.find('[name="weight"]').at(0).simulate('change', { target: { name: 'weight', value: '123@#$' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because height contains letters', () => {
        wrapper.find('[name="height"]').at(0).simulate('change', { target: { name: 'height', value: '123abc' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile because height contains special characters', () => {
        wrapper.find('[name="height"]').at(0).simulate('change', { target: { name: 'height', value: '123@#$' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(0);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })

    it('handleSubmit should not push to profile when cannot fetch userData because email is already in use', () => {
        let mockProps = {
            userDataReducer: {
                id: 0,
                name: '',
                surname: '',
                email: '',
                password: '',
                gender: '',
                birthday: '',
                weight: '',
                height: '',
                isPending: false
            },
            // onRegister: jest.fn().mockRejectedValueOnce("ERROR: the email is already in use"),
            onRegister: jest.fn().mockResolvedValueOnce("ERROR: the email is already in use"),
            onLogout: jest.fn()
          };
        wrapper = shallow(<RegisterMainPage {...mockProps} history={historyMock}/>);
        wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: 'julian' } });
        wrapper.find('[name="surname"]').at(0).simulate('change', { target: { name: 'surname', value: 'livrone' } });
        wrapper.find('[name="email"]').at(0).simulate('change', { target: { name: 'email', value: 'julianlivrone@gmail.com' } });
        wrapper.find('[name="password"]').at(0).simulate('change', { target: { name: 'password', value: 'asd' } });
        wrapper.find('[name="gender"]').at(0).simulate('change', { target: { name: 'gender', value: 'M' } });
        wrapper.find('[name="date"]').at(0).simulate('change', 'Sat Sep 18 2021 11:00:00 GMT-0300 (hora estándar de Argentina)');
        wrapper.find('[name="weight"]').at(0).simulate('change', { target: { name: 'weight', value: '90' } });
        wrapper.find('[name="height"]').at(0).simulate('change', { target: { name: 'height', value: '170' } });
        wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);
        expect(wrapper.instance().props.onRegister).toHaveBeenCalledTimes(1);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(0);
    })
})