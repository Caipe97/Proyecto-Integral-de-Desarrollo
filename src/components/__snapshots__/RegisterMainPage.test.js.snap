// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterMainPage tests renders RegisterMainPage without crashing 1`] = `
<div
  className="contenedorRe"
>
  <Header
    history={
      Object {
        "push": [MockFunction],
      }
    }
    onLogout={[MockFunction]}
  />
  <div
    className="contenidoRe"
  >
    <div
      className="col1"
    >
      <img
        alt="imagen tracking"
        src="2pages.jpg"
        style={
          Object {
            "height": 240,
            "width": 240,
          }
        }
      />
      <h1
        style={
          Object {
            "left": "49%",
          }
        }
      >
        Nutriva
      </h1>
    </div>
    <div
      className="col2"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      <p>
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </div>
  </div>
  <div
    className="sidebar"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <div
        className="box"
        style={Object {}}
      >
        <img
          alt="imagen tracking"
          src="avatar.png"
          style={
            Object {
              "width": 100,
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "marginTop": 45,
          }
        }
      >
        <h1
          className="f1"
        >
          RegisterMainPage
        </h1>
      </div>
      <form
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "center",
          }
        }
      >
        <WithStyles(ForwardRef(TextField))
          label="Name"
          name="name"
          onChange={[Function]}
          required={true}
          type="name"
          value=""
        />
        <WithStyles(ForwardRef(TextField))
          label="Surname"
          name="surname"
          onChange={[Function]}
          required={true}
          type="surname"
          value=""
        />
        <WithStyles(ForwardRef(TextField))
          label="Email"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
        <WithStyles(ForwardRef(TextField))
          label="Password"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
        <div
          style={
            Object {
              "margin": 10,
            }
          }
        >
          <input
            id="masculino"
            name="gender"
            onChange={[Function]}
            type="radio"
            value="M"
          />
            
          <label>
            Masculino
          </label>
            
          <input
            id="femenino"
            name="gender"
            onChange={[Function]}
            type="radio"
            value="F"
          />
            
          <label>
            Femenino
          </label>
        </div>
        <r
          allowSameDay={false}
          customTimeInput={null}
          dateFormat="dd-MM-yyyy h:mm aa"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          enableTabLoop={true}
          excludeScrollbar={true}
          focusSelectedMonth={false}
          monthsShown={1}
          name="date"
          nextMonthButtonLabel="Next Month"
          nextYearButtonLabel="Next Year"
          onBlur={[Function]}
          onCalendarClose={[Function]}
          onCalendarOpen={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          placeholderText="birthday"
          preventOpenOnFocus={false}
          previousMonthButtonLabel="Previous Month"
          previousYearButtonLabel="Previous Year"
          readOnly={false}
          renderDayContents={[Function]}
          selected=""
          shouldCloseOnSelect={true}
          showFourColumnMonthYearPicker={false}
          showFullMonthYearPicker={false}
          showMonthYearPicker={false}
          showPopperArrow={true}
          showPreviousMonths={false}
          showQuarterYearPicker={false}
          showTimeInput={false}
          showTimeSelect={true}
          showTwoColumnMonthYearPicker={false}
          showYearPicker={false}
          strictParsing={false}
          timeCaption="Time"
          timeInputLabel="Time"
          timeIntervals={30}
          withPortal={false}
          yearItemNumber={12}
        />
        <WithStyles(ForwardRef(TextField))
          label="Weight"
          name="weight"
          onChange={[Function]}
          required={true}
          type="weight"
          value=""
        />
        <WithStyles(ForwardRef(TextField))
          label="Height"
          name="height"
          onChange={[Function]}
          required={true}
          type="height"
          value=""
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Registrarse
        </button>
        <p
          style={
            Object {
              "marginBlock": "0em",
              "marginTop": "10%",
            }
          }
        >
          Si ya tenes una cuenta?
        </p>
        <Link
          style={
            Object {
              "color": "black",
            }
          }
          to="/"
        >
          Login
        </Link>
      </form>
    </div>
  </div>
  <Footer />
</div>
`;
